app/helpers/taxonomy_helper.rb

```diff
@@ -135,4 +135,53 @@ def taxonomy_selects(f, selected_ids, taxonomy, label, options = {}, options_htm
     options[:label]    ||= _(label)
     multiple_selects f, label.downcase, taxonomy.authorized("assign_#{label.downcase}", taxonomy), selected_ids, options, options_html
   end
+
+  def all_checkbox(f, resource)
+    return unless User.current.admin? || User.current.filters.joins(:permissions).where({:'permissions.name' => "view_#{resource}",
+                                                       :search => nil,
+                                                       :taxonomy_search => nil}).present?
+    checkbox_f(f, :ignore_types,
+                 {:label => translated_label(resource, :all),
+                  :multiple => true,
+                  :onchange => 'ignore_checked(this)'},
+                  resource.to_s.humanize)
+  end
+
+  def show_resource_if_allowed(f, taxonomy, resource_options)
+    if resource_options.is_a? Hash
+      resource = resource_options[:resource]
+      association = resource_options[:association]
+    else
+      resource = resource_options
+      association = resource.to_s.classify.constantize
+    end
+    return unless User.current.allowed_to?("view_#{resource}".to_sym)
+    ids = "#{association.table_name.singularize}_ids".to_sym
+
+    content_tag(:div, :id => resource, :class => "tab-pane") do
+      all_checkbox(f, resource) +
+      multiple_selects(f, association.table_name.to_sym, association, taxonomy.selected_or_inherited_ids[ids],
+                           {:disabled => taxonomy.used_and_selected_or_inherited_ids[ids],
+                            :label => translated_label(resource, :select)},
+                           {'data-mismatches' => taxonomy.need_to_be_selected_ids[ids].to_json,
+                            'data-inheriteds' => taxonomy.inherited_ids[ids].to_json,
+                            'data-useds' => taxonomy.used_ids[ids].to_json })
+    end
+  end
+
+  def translated_label(resource, verb)
+    verb = verb == :all ?  0 : 1
+    labels = { :users => [_("All Users"), _("Select Users")],
+               :smart_proxies => [_("All Smart Proxies"), _("Select Smart Proxies")],
+               :subnets => [_("All Subnets"), _("Select Subnets")],
+               :compute_resources => [_("All Compute Resources"), _("Select Compute Resources")],
+               :media => [_("All Media"), _("Select Media")],
+               :templates => [_("All Templates"), _("Select Templates")],
+               :domains => [_("All Domains"), _("Select Domains")],
+               :realms => [_("All Realms"), _("Select Realms")],
+               :environments => [_("All Environments"), _("Select Environments")],
+               :hostgroups => [_("All Hostgroups"), _("Select Hostgroups")],
```

Comment:
"host group", not "hostgroup"

You've also changed the capitalisation throughout, I'd suggest they should be sentence case.